### ISSUE TITLE
Slight deviations from reported metric

### ISSUE THREAD

**Issue by TillBeemelmanns** [yinjunbo/IS-Fusion#17]
Hi, 

I tested IS-Fusion on the nuScenes validation split and I do not get the exact same metrics as reported. The difference is small  0.7365 vs. 0.740 NDS, however constant across several runs. Any idea why ?

Here is the log:
```
python tools/test.py /workspace/configs/isfusion/isfusion_0075voxel.py /workspace/ckpts/IS-Fusion_epoch_10.pth --eval bbox
load checkpoint from local path: /workspace/ckpts/IS-Fusion_epoch_10.pth
[                                                  ] 0/6019, elapsed: 0s, ETA:/opt/conda/lib/python3.7/site-packages/torch/nn/functional.py:4004: UserWarning: Default grid_sample and affine_grid behavior has changed to align_corners=False since 1.3.0. Please specify align_corners=True if the old behavior is desired. See the documentation of grid_sample for details.
  "Default grid_sample and affine_grid behavior has changed "
drop_info is set to {0: {'max_tokens': 36, 'drop_range': (0, 100000)}}, in input_layer
/workspace/mmdet3d/ops/sst/sst_ops.py:249: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  win_coors_x = shifted_coors_x // win_shape_x
/workspace/mmdet3d/ops/sst/sst_ops.py:250: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  win_coors_y = shifted_coors_y // win_shape_y
/workspace/mmdet3d/ops/sst/sst_ops.py:251: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  win_coors_z = shifted_coors_z // win_shape_z
/workspace/mmdet3d/models/sst/sst_input_layer_v2.py:258: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  inv_freq = self.pos_temperature ** (2 * (inv_freq // 2) / pos_length)
/workspace/mmdet3d/ops/sst/sst_ops.py:91: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  num_windows = (this_inds // max_tokens).max().item() + 1
drop_info is set to {0: {'max_tokens': 36, 'drop_range': (0, 100000)}}, in input_layer
/workspace/mmdet3d/models/dense_heads/transfusion_head_v2.py:825: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  top_proposals_class = top_proposals // heatmap.shape[-1]
[                                                  ] 1/6019, 0.5 task/s, elapsed: 2s, ETA: 11557s/workspace/mmdet3d/core/bbox/coders/transfusion_bbox_coder.py:99: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  self.post_center_range, device=heatmap.device)
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6019/6019, 2.5 task/s, elapsed: 2370s, ETA:     0s
Formating bboxes of pts_bbox
Start to convert detection format...
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 6019/6019, 24.9 task/s, elapsed: 242s, ETA:     0s
Results writes to /tmp/tmp24w7qwk7/results/pts_bbox/results_nusc.json
Evaluating bboxes of pts_bbox
mAP: 0.7250                                                                                                                                                                           
mATE: 0.2657
mASE: 0.2516
mAOE: 0.2929
mAVE: 0.2646
mAAE: 0.1850
NDS: 0.7365
Eval time: 150.6s

Per-class results:
Object Class    AP      ATE     ASE     AOE     AVE     AAE
car     0.894   0.162   0.150   0.053   0.266   0.194
truck   0.672   0.307   0.179   0.068   0.245   0.234
bus     0.787   0.313   0.181   0.066   0.456   0.250
trailer 0.517   0.482   0.201   0.454   0.193   0.141
construction_vehicle    0.368   0.661   0.414   1.008   0.144   0.288
pedestrian      0.908   0.125   0.287   0.340   0.213   0.104
motorcycle      0.806   0.180   0.237   0.214   0.385   0.255
bicycle 0.697   0.147   0.256   0.381   0.216   0.015
traffic_cone    0.846   0.099   0.325   nan     nan     nan
barrier 0.755   0.182   0.285   0.052   nan     nan
```

I am using this docker setup and the provided weights for IS-Fusion.
```Docker
ARG PYTORCH="1.10.0"
ARG CUDA="11.3"
ARG CUDNN="8"

FROM pytorch/pytorch:${PYTORCH}-cuda${CUDA}-cudnn${CUDNN}-devel

ENV TORCH_CUDA_ARCH_LIST="6.0 6.1 7.0 7.5 8.0 8.6+PTX" \
    TORCH_NVCC_FLAGS="-Xfatbin -compress-all" \
    CMAKE_PREFIX_PATH="$(dirname $(which conda))/../" \
    FORCE_CUDA="1"

# Avoid Public GPG key error
# https://github.com/NVIDIA/nvidia-docker/issues/1631
RUN rm /etc/apt/sources.list.d/cuda.list \
    && rm /etc/apt/sources.list.d/nvidia-ml.list \
    && apt-key del 7fa2af80 \
    && apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub \
    && apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/7fa2af80.pub

# (Optional, use Mirror to speed up downloads)
# RUN sed -i 's/http:\/\/archive.ubuntu.com\/ubuntu\//http:\/\/mirrors.aliyun.com\/ubuntu\//g' /etc/apt/sources.list && \
#    pip config set global.index-url https://pypi.tuna.tsinghua.edu.cn/simple

# Install the required packages
RUN apt-get update \
    && apt-get install -y vim htop screen ffmpeg libsm6 libxext6 git ninja-build libglib2.0-0 libsm6 libxrender-dev libxext6 \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/*

# Install MMEngine, MMCV and MMDetection
RUN pip install openmim && \
    mim install "mmengine" "mmcv==1.4.0" "mmdet==2.14.0" "mmsegmentation==0.14.1" \
    && pip install gpustat spconv-cu113 ipdb

# Install MMDetection3D
RUN conda clean --all 
```

Then in the docker environment I run
```bash
# Install mmdetection3d
echo "Installing mmdetection3d"
pip install --no-cache-dir -e .

echo "Installing TorchEx"
cd mmdet3d/ops/TorchEx && pip install -v .
```

**Comment by rubbish001** [comment]
兄弟，有卡吗？合作搞一篇多帧融合的怎么样，提供卡就可以挂名，冲CVPR25怎么样

**Comment by yinjunbo** [comment]
@TillBeemelmanns, thank you for your interest in our work. In the released code, we have omitted the point-to-region transformer to save memory and ensure the model can be trained by v100 GPUs which are more available to researchers (we used 8*A100-80G in the original paper), thus the performance slightly deviates from the paper.

**Comment by rubbish001** [comment]
真羡慕你们这些a100的，我4卡4090刷的，nnd，test集mAP 73.9%和 NDS 74.7%（比你低一点），单帧图片搞的，准备投aaai24

**Comment by TillBeemelmanns** [comment]
Thanks for clarification. I added your here published IS-Fusion model to our MultiCorrupt robustness benchmark https://github.com/ika-rwth-aachen/MultiCorrupt
Maybe you are interested in this. 

Best,
Till
